LLK/scratch-vm
# Handle unknown opcode in input

### Resolves

https://github.com/LLK/scratch-vm/issues/1682

### Proposed Changes

Fix import error when block has an input with an unknown opcode.

### Reason for Changes

#1254 skips unknown opcodes when importing blocks from sb2 projects. However, that PR missed the case where an input is undefined. This PR fixes projects that were failing to load because of that issue.

### Test Coverage

This is my first issue, I'm not sure how to test other than manually with this project: https://llk.github.io/scratch-gui/develop/#116599254

I added a unit test called "unknown-opcode-as-reporter-block.js"

### How to test this

1. In Scratch 2.0, drag out a block with an input
2. Put a block with unknown opcodes in the input. Examples are the "picoboard" extension blocks, such as "sensor value"
3. Import the project into 3.0
4. The project should be able to load and there will be a default value (shadow) instead of the input. There should be warnings about skipped opcodes in the console.
